### PR TITLE
SPLAT-1638: Added multi vCenter support

### PR DESCRIPTION
**Pull request by vr4manta**
[SPLAT-1638](https://issues.redhat.com//browse/SPLAT-1638)

### Changes
- Bump API to get VSphereMultiVCenters feature gate
- Enhanced operator to use new YAML vSphere cloud config

### Dependencies
- https://github.com/openshift/api/pull/1842
- https://github.com/openshift/machine-api-operator/pull/1234
- https://github.com/openshift/machine-api-operator/pull/1250

### Notes
I currently have same changes as https://github.com/openshift/machine-api-operator/pull/1250/files due to hitting the same things earlier.  Talked to Joel and having these done as separate PR.

### CODE REVIEW

**Comment by vr4manta** [comment]
/hold
Looking at upgrade scenarios.

File: hack/crds-sync.sh [hunk 1]
```diff
@@ -11,4 +11,4 @@ for crd in "${CRDS_MAPPING[@]}" ; do
     SRC="${crd%%:*}"
     DES="${crd##*:}"
     cp "$SRC" "install/$DES"
-done
+done
```
Comment (JoelSpeed) [inline hack/crds-sync.sh:14]: Please revert

File: pkg/controller/vsphere/actuator_test.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ func TestMachineEvents(t *testing.T) {
 	defer server.Close()
 
 	host, port, err := net.SplitHostPort(server.URL.Host)
+	fmt.Printf("Host: %v Port: %v\n", host, port)
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/actuator_test.go:51]: Debug?

File: pkg/controller/vsphere/actuator_test.go [hunk 3]
```diff
@@ -100,9 +101,7 @@ func TestMachineEvents(t *testing.T) {
 		},
 	}
 	g.Expect(k8sClient.Create(context.Background(), configNamespace)).To(Succeed())
-	defer func() {
-		g.Expect(k8sClient.Delete(context.Background(), configNamespace)).To(Succeed())
-	}()
+	// Cannot delete openshift-config so not setting up defer to remove it
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/actuator_test.go:104]: I think the name of the namespace being used was changed, so perhaps we can revert this section

File: pkg/controller/vsphere/actuator_test.go [hunk 4]
```diff
@@ -132,7 +132,8 @@ func TestMachineEvents(t *testing.T) {
 		g.Expect(k8sClient.Delete(context.Background(), &credentialsSecret)).To(Succeed())
 	}()
 
-	testConfig := fmt.Sprintf(testConfigFmt, port)
+	testConfig := fmt.Sprintf(testConfigFmt, port, credentialsSecretName, testNamespaceName)
+	fmt.Printf("Config: %v\n", testConfig)
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/actuator_test.go:136]: Debug?

File: pkg/controller/vsphere/util_test.go [hunk 5]
```diff
@@ -15,19 +15,22 @@ const (
 	testRegion       = "testRegion"
 	testZone         = "testZone"
 	testPort         = "443"
-	testInsecureFlag = "1"
+	testInsecureFlag = true
```
Comment (JoelSpeed) [inline pkg/controller/vsphere/util_test.go:18]: Is there a change in the accepted values?

File: pkg/controller/vsphere/actuator_test.go (hunk 3 above)
Comment (vr4manta) [reply pkg/controller/vsphere/actuator_test.go:104]: Ok, I'll clean up and retry.

File: pkg/controller/vsphere/util_test.go (hunk 5 above)
Comment (vr4manta) [reply pkg/controller/vsphere/util_test.go:18]: no, its a boolean that comes out of the object, not a "1".  below you can see "1" is still used in creating ini file.

File: pkg/controller/vsphere/actuator_test.go (hunk 2 above)
Comment (vr4manta) [reply pkg/controller/vsphere/actuator_test.go:51]: I can remove this. I like this output so when test failed, i can verify the port number that was being used matched.

File: hack/crds-sync.sh (hunk 1 above)
Comment (vr4manta) [reply hack/crds-sync.sh:14]: Good catch, i thought i removed that.  My IDE annoys me sometimes.

File: pkg/controller/vsphere/actuator_test.go (hunk 3 above)
Comment (vr4manta) [reply pkg/controller/vsphere/actuator_test.go]: Looks good, I removed this change now.

File: pkg/controller/vsphere/actuator_test.go (hunk 4 above)
Comment (vr4manta) [reply pkg/controller/vsphere/actuator_test.go]: Removed debug.

File: pkg/controller/vsphere/util.go [hunk 6]
```diff
@@ -169,18 +133,34 @@ func conditionFailed() metav1.Condition {
 	}
 }
 
-func getPortFromConfig(config *vSphereConfig) string {
+func getVCenterPortFromConfig(config *vsphere.Config, vcenter string) string {
 	if config != nil {
-		return config.Global.Port
+		//return config.Global.Port
```
Comment (jcpowermac) [inline pkg/controller/vsphere/util.go:138]: remove comment?

Comment (vr4manta) [reply pkg/controller/vsphere/util.go:138]: Good catch.  I'll update that.

**Pull request merged**